client/src/modules/Score/components/ScoreTable/index.tsx

```diff
@@ -65,8 +65,12 @@ export function ScoreTable(props: Props) {
         courseService.getCourseScore(students.pagination, filters, students.orderBy),
         courseService.getCourseTasks(),
       ]);
-      const sortedTasks = courseTasks.filter(task => !!task.studentEndDate || props.course.completed);
-
+      const sortedTasks = courseTasks
+        .filter(task => !!task.studentEndDate || props.course.completed)
+        .map(task => {
+          task.isVisible = !notVisibleColumns?.includes(task.name);
+          return task;
+        });
```

Comment:
Hope you're know what are you doing.
If there weren't be filter then you would mutate original `courseTasks` array.

Reply:
done